scripts/capture_pane_helpers.sh

```diff
@@ -1,26 +1,15 @@
-# Variables in this helper should be set in the file sourcing the helper.
-# Required variables:
-# - path_option & default_path
-# - name_option & default_name
-
-get_path() {
-	get_tmux_option "$path_option" "$default_path"
-}
-
-# `tmux save-buffer` command does not perform interpolation, so we're doing it
-# "manually" with `display-message`
-get_filename() {
-	local name_template=$(get_tmux_option "$name_option" "$default_name")
-	tmux display-message -p "$name_template"
-}
-
 capture_pane() {
-	local file="$(get_path)/$(get_filename)"
 	local capture_scope=$1
 	if [ $capture_scope == "History" ]; then
+		local file=$(tmux display-message -p "$save_complete_history_full_filename")
+		local path=$(tmux display-message -p "$save_complete_history_path")
+		mkdir -p "$path"
 		local history_limit="$(tmux display-message -p -F "#{history_limit}")"
 		tmux capture-pane -J -S "-${history_limit}" -p > "$file"
 	elif [[ $capture_scope == "Screen capture" ]]; then
+		local file=$(tmux display-message -p "$screen_capture_full_filename")
+		local path=$(tmux display-message -p "$screen_capture_path")
+		mkdir -p "$path"
```

Comment:
After your refactoring it seems as this function is unnecessarily doing 2 things: capturing history and doing screen capture.

Would it make things simpler if this one was split into two functions? With that we could get rid of this whole file because the functions would be kept in files where they're used..

Reply:
I agree it will be simpler than the current state. I will refactor this part